src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol

```diff
@@ -0,0 +1,184 @@
+  }
+
+  error FailedToSendMATIC();
+
+  IAavePolEthERC20Bridge public constant bridge =
+    IAavePolEthERC20Bridge(MiscPolygon.AAVE_POL_ETH_BRIDGE);
+
+  IAavePolEthPlasmaBridge public constant plasmaBridge =
+    IAavePolEthPlasmaBridge(0xc980508cC8866f726040Da1C0C61f682e74aBc39);
+
+  uint256 public constant A_WMATIC_AMOUNT_REIMBURSEMENT = 4000 ether;
+  address public constant BGD_RECIPIENT = 0xbCEB4f363f2666E2E8E430806F37e97C405c130b;
+  address public constant NATIVE_MATIC = 0x0000000000000000000000000000000000001010;
+
+  function execute() external {
+    AaveV2Polygon.COLLECTOR.transfer(
+      AaveV2PolygonAssets.WMATIC_A_TOKEN,
+      BGD_RECIPIENT,
+      A_WMATIC_AMOUNT_REIMBURSEMENT
+    );
+
+    _migrateV2ToV3();
+    _withdrawV2Tokens();
+    _withdrawV3Tokens();
+    _withdrawMatic();
+
+    bridge.bridge(
+      AaveV3PolygonAssets.USDC_UNDERLYING,
+      IERC20(AaveV3PolygonAssets.USDC_UNDERLYING).balanceOf(address(bridge))
+    );
+
+    bridge.bridge(
+      AaveV3PolygonAssets.DAI_UNDERLYING,
+      IERC20(AaveV3PolygonAssets.DAI_UNDERLYING).balanceOf(address(bridge))
+    );
+
+    plasmaBridge.bridge(IERC20(NATIVE_MATIC).balanceOf(address(plasmaBridge)));
+  }
+
+  function _migrateV2ToV3() internal {
+    AaveV3Polygon.COLLECTOR.transfer(
+      AaveV2PolygonAssets.USDT_A_TOKEN,
+      address(this),
+      IERC20(AaveV2PolygonAssets.USDT_A_TOKEN).balanceOf(address(AaveV3Polygon.COLLECTOR)) - 1e6
+    );
+
+    AaveV2Polygon.POOL.withdraw(
+      AaveV2PolygonAssets.USDT_UNDERLYING,
+      type(uint256).max,
+      address(this)
+    );
+
+    uint256 amount = IERC20(AaveV2PolygonAssets.USDT_UNDERLYING).balanceOf(address(this));
+    IERC20(AaveV2PolygonAssets.USDT_UNDERLYING).forceApprove(address(AaveV3Polygon.POOL), amount);
+
+    AaveV3Polygon.POOL.deposit(
+      AaveV2PolygonAssets.USDT_UNDERLYING,
+      amount,
+      address(AaveV3Polygon.COLLECTOR),
+      0
+    );
+  }
+
+  function _withdrawMatic() internal {
+    AaveV3Polygon.COLLECTOR.transfer(
+      AaveV2PolygonAssets.WMATIC_A_TOKEN,
+      address(this),
+      IERC20(AaveV2PolygonAssets.WMATIC_A_TOKEN).balanceOf(address(AaveV3Polygon.COLLECTOR)) -
+        1 ether
+    );
+
+    AaveV2Polygon.POOL.withdraw(
+      AaveV2PolygonAssets.WMATIC_UNDERLYING,
+      type(uint256).max,
+      address(this)
+    );
+
+    AaveV3Polygon.COLLECTOR.transfer(
+      AaveV3PolygonAssets.WMATIC_A_TOKEN,
+      address(this),
+      IERC20(AaveV3PolygonAssets.WMATIC_A_TOKEN).balanceOf(address(AaveV3Polygon.COLLECTOR)) -
+        1 ether
+    );
+
+    AaveV3Polygon.POOL.withdraw(
+      AaveV3PolygonAssets.WMATIC_UNDERLYING,
+      type(uint256).max,
+      address(this)
+    );
+
+    IWMatic(AaveV3PolygonAssets.WMATIC_UNDERLYING).withdraw(
+      IERC20(AaveV3PolygonAssets.WMATIC_UNDERLYING).balanceOf(address(this))
+    );
+
+    (bool ok, ) = address(plasmaBridge).call{value: address(this).balance}('');
+
+    if (!ok) revert FailedToSendMATIC();
+  }
```

Comment:
```suggestion
```
@efecarranza I would not send funds to the plasma bridge, as we will do the MATIC bridging in second part.

Reply:
Just one remark, Polygon plasma bridge has a log cap protection of 10 logs in a bridge transaction request.  Transferring funds to the bridge contract in Part A makes sense to me to avoid this issue during Part B.  

Currently not sure if it will be a problem or not since this is the first time we are doing it in production.